File: src/main/java/org/cloudfoundry/credhub/entity/PasswordCredentialVersionData.java
package org.cloudfoundry.credhub.entity;

import org.hibernate.annotations.NotFound;
import org.hibernate.annotations.NotFoundAction;

import javax.persistence.CascadeType;
import javax.persistence.DiscriminatorValue;
import javax.persistence.Entity;
import javax.persistence.JoinColumn;
import javax.persistence.OneToOne;
import javax.persistence.PrimaryKeyJoinColumn;
import javax.persistence.SecondaryTable;

@Entity
@DiscriminatorValue(PasswordCredentialVersionData.CREDENTIAL_TYPE)
@SecondaryTable(
    name = PasswordCredentialVersionData.TABLE_NAME,
    pkJoinColumns = {@PrimaryKeyJoinColumn(name = "uuid", referencedColumnName = "uuid")}
)
public class PasswordCredentialVersionData extends CredentialVersionData<PasswordCredentialVersionData> {

  public static final String CREDENTIAL_TYPE = "password";
  static final String TABLE_NAME = "password_credential";

  @OneToOne(cascade = CascadeType.ALL)
  @NotFound(action = NotFoundAction.IGNORE)
  @JoinColumn(table = PasswordCredentialVersionData.TABLE_NAME, name = "password_parameters_uuid")
  private EncryptedValue encryptedGenerationParameters;

  @SuppressWarnings("unused")
  public PasswordCredentialVersionData() {
  }

  public PasswordCredentialVersionData(String name) {
    super(name);
  }


  public PasswordCredentialVersionData setEncryptedGenerationParameters(
     EncryptedValue encryptedGenerationParameters) {
    this.encryptedGenerationParameters = encryptedGenerationParameters;
    return this;
  }

  public EncryptedValue getEncryptedGenerationParameters() {
    return encryptedGenerationParameters;
  }

  @Override
  public String getCredentialType() {
    return CREDENTIAL_TYPE;
  }
}

File: components/credentials/src/main/kotlin/org/cloudfoundry/credhub/entity/PasswordCredentialVersionData.kt
package org.cloudfoundry.credhub.entity

import jakarta.persistence.CascadeType
import jakarta.persistence.DiscriminatorValue
import jakarta.persistence.Entity
import jakarta.persistence.JoinColumn
import jakarta.persistence.OneToOne
import jakarta.persistence.PrimaryKeyJoinColumn
import jakarta.persistence.SecondaryTable
import org.cloudfoundry.credhub.entities.EncryptedValue
import org.hibernate.annotations.NotFound
import org.hibernate.annotations.NotFoundAction

@Entity
@DiscriminatorValue(PasswordCredentialVersionData.CREDENTIAL_TYPE)
@SecondaryTable(
    name = PasswordCredentialVersionData.TABLE_NAME,
    pkJoinColumns = [PrimaryKeyJoinColumn(name = "uuid", referencedColumnName = "uuid")],
)
class PasswordCredentialVersionData : CredentialVersionData<PasswordCredentialVersionData> {
    @OneToOne(cascade = [CascadeType.ALL])
    @NotFound(action = NotFoundAction.IGNORE)
    @JoinColumn(table = TABLE_NAME, name = "password_parameters_uuid")
    var encryptedGenerationParameters: EncryptedValue? = null

    override val credentialType: String
        get() = CREDENTIAL_TYPE

    constructor() : super()

    constructor(name: String) : super(name)

    companion object {
        const val CREDENTIAL_TYPE = "password"
        const val TABLE_NAME = "password_credential"
    }
}
